translator/source_code/combine_wzdx.py

```diff
@@ -0,0 +1,157 @@
+import json
+import copy
+from shapely.geometry.polygon import Polygon
+from shapely.geometry import Point
+import geopy
+from geopy.distance import geodesic
+import pyproj
+import getopt
+import sys
+
+
+def main():
+
+    icone_file, cotrip_file, outputfile = parse_arguments(
+        sys.argv[1:], default_output_file_name='combined_wzdx_message.geojson')
+
+    with open(outputfile, 'w+') as f:
+        wzdx_icone = json.loads(open(
+            icone_file, 'r').read())
+        wzdx_cotrip = json.loads(open(
+            cotrip_file, 'r').read())
+
+        polygon = generate_polygon(
+            wzdx_cotrip['features'][0]['geometry']['coordinates'], 100)
+        feature = iterate_feature(polygon, wzdx_icone)
+        if feature:
+
+            f.write(json.dumps(combine_wzdx(
+                wzdx_cotrip, wzdx_icone, feature), indent=2))
+            print('duplicate iCone and COtrip messages were found and combined WZDx message was written to combined_wzdx_message.geojson file')
+
+        else:
+            print('no duplicate messages were found')
+
+
+def combine_wzdx(wzdx_cotrip, wzdx_icone, icone_feature):
```

Comment:
With this combine function, it looks like we are just taking everything from COtrip, then adding a single feature `vehicle_impact` value. That seems limited for a combined stream, is it possible there's more data in iCone than we are taking here? What about differing geometry for instance - could we have iCone with multiple points outside the polygon defined above or are we always getting a single point with iCone?

Reply:
The iCone message really doesn't have a ton of information, compared to the cotrip/salesforce message. There are only a few fields that are more informative in iCone than in COtrip, and this is (currently) just the vehicle impact

Reply:
I believe we are iterating over the iCone geometry to catch any points that are in the polygon, on line 36

Reply:
Line numbers are a bit screwy now with all the changes going on...are you meaning the `iterate_feature` method? It looks like we take in the polygon generated from the COtrip data an an iCone message. Then we iterate over the features in iCone, look to the coordinates, and determine if any coordinates are in the polygon. We then return the first feature we find with a single point contained in the polygon. This could leave out additional features (with the additional vehicle impacts), or points not in the polygon if any such points exist.

Reply:
ok I see what you are saying now. You are correct, we stop looking after a single feature matches. **We made the assumption that duplicated workzones would not appear in a single iCone dataset**, I think that is a valid assumption. However, when this gets expanded to more datasets, this could potentially be a problem. I will leave it up to @payneBrandon to decide whether it is worth fixing

Reply:
@payneBrandon  should be fixed